sqlitecluster/SQLite.cpp

```diff
@@ -819,9 +821,15 @@ int SQLite::commit(const string& description, function<void()>* preCheckpointCal
             if (_sharedData.outstandingFramesToCheckpoint) {
                 auto start = STimeNow();
                 int framesCheckpointed = 0;
-                sqlite3_wal_checkpoint_v2(_db, 0, SQLITE_CHECKPOINT_PASSIVE, NULL, &framesCheckpointed);
+
+                // For non-passive checkpoints, we must set a busy timeout in order to wait on any readers.
+                // We set it to 2 minutes as the majority of transactions should take less than that.
+                if (_checkpointMode != SQLITE_CHECKPOINT_PASSIVE) {
+                    sqlite3_busy_timeout(_db, 120'000);
+                }
```

Comment:
Actually shouldn't we move this whole block to `commonConstructorInitialization` with the rest of the setup code where we set handlers? Why is it here?

Reply:
I guess we're trying to set and then unset it for each checkpoint, but what's the downside to setting it once and leaving it set?

Reply:
Good point, I asked Dan.

Reply:
Based on the email discussion, I think we can just set this at construction.

Reply:
Updated with the change